src/pragmaUtil.ts

```diff
@@ -0,0 +1,249 @@
+  switch (osName.toLowerCase()) {
+    case "windows":
+      return OsType.Windows;
+    case "linux":
+      return OsType.Linux;
+    case "mac":
+      return OsType.Mac;
+  }
+}
+
+/**
+ * Comment/Uncomment lines if matches OS name or Hostname.
+ * Usage: @sync os=[OS name] host=[hostName]
+ * Notes: Hostname must be defined in sync.host setting. It could be used for parse any JSON valid string.
+ * @export
+ * @class PragmaUtil
+ */
+export default class PragmaUtil {
+  /**
+   * Process @sync pragma statements before file is saved.
+   * Comment lines that don't match with the user OS or host value.
+   * @static
+   * @param {string} settingsContent a valid JSON string
+   * @returns {string}
+   * @memberof PragmaUtil
+   */
+  public static processBeforeWrite(
+    settingsContent: string,
+    osType: OsType,
+    hostName: string
+  ): string {
+    let result: string = settingsContent;
+
+    const pragmaSettingsBlocks: RegExpMatchArray = result.match(
+      this.PragmaRegExp
+    );
+
+    if (pragmaSettingsBlocks !== null) {
+      for (const block of pragmaSettingsBlocks) {
+        // line e.g.: // @sync os=windows host=Laptop\n"window.menuBarVisibility": "none",
+        try {
+          // check OS pragma
+          const osMatch: RegExpMatchArray = block.match(/os=(\w+)/);
+          if (osMatch !== null) {
+            const osFromPragma = osMatch[1].toLowerCase();
+
+            if (!SUPPORTED_OS.includes(osFromPragma)) {
+              continue;
+            }
+            if (GetOsEnum(osFromPragma) !== osType) {
+              result = result.replace(block, this.commentLineAfterBreak(block));
+              continue; // no need to lookup the host name
+            }
+          }
+
+          // check host pragma
+          const hostMatch: RegExpMatchArray = block.match(/host=(\S+)/);
+          if (hostMatch !== null) {
+            const hostFromPragma = hostMatch[1];
+            if (
+              hostName === null ||
+              hostName === "" ||
+              hostFromPragma.toLowerCase() !== hostName.toLowerCase()
+            ) {
+              result = result.replace(block, this.commentLineAfterBreak(block));
+              continue;
+            }
+          }
+
+          // check env pragma
+          const envMatch: RegExpMatchArray = block.match(/env=(\S+)/);
+          if (envMatch !== null) {
+            const envFromPragma = envMatch[1];
+            if (process.env[envFromPragma.toUpperCase()] === undefined) {
+              result = result.replace(block, this.commentLineAfterBreak(block));
+              continue;
+            }
+          }
+
+          // if os, host and evn matched the current machine make sure to uncomment the setting
+          result = result.replace(block, this.uncommentLineAfterBreak(block));
+        } catch (e) {
+          continue;
+        }
+      }
+    }
+
+    result = this.removeIgnoreBlocks(result);
+
+    return result;
```

Comment:
Since the text itself isn't a valid JSON because all the slashes the only way to ensure is a valid json is by removing all the comments. Do you agree? You can put /* */ comments on the settings.json and it won't trigger an error. The only solution I think could do the work is get all the settings that matches `"\S+":.+,?\n*`. Do you have a better idea?